src/main/java/com/chekanova/imagetool/service/comparison/impl/ImageComparisonServiceImpl.java

```diff
@@ -0,0 +1,193 @@
+        }
+        for (int y = minY; y <= maxY; y++) {
+            image.setRGB(minX, y, Color.red.getRGB());
+            image.setRGB(maxX, y, Color.red.getRGB());
+        }
+    }
+
+    /**
+     * Returns maximum difference from red, blue and green colors.
+     * @param firstImageColor  pixel color of first image
+     * @param secondImageColor pixel color of second image
+     * @return double value of maximum difference between different colors between pixels
+     */
+    private double calculateColorDifference(Color firstImageColor, Color secondImageColor) {
+        int r1 = firstImageColor.getRed();
+        int r2 = secondImageColor.getRed();
+        int g1 = firstImageColor.getGreen();
+        int g2 = secondImageColor.getGreen();
+        int b1 = firstImageColor.getBlue();
+        int b2 = secondImageColor.getBlue();
+        double redDifference = Math.abs(r1 - r2) * 100.0 / Math.min(r1, r2);
+        double greenDifference = Math.abs(g1 - g2) * 100.0 / Math.min(g1, g2);
+        double blueDifference = Math.abs(b1 - b2) * 100.0 / Math.min(b1, b2);
+        return Math.max(redDifference, Math.max(greenDifference, blueDifference));
+    }
+
+    private void clearProcessedPixels(Box box, boolean[][] comparisonResult) {
+        for (int x = box.minX; x <= box.maxX; x++) {
+            for (int y = box.minY; y <= box.maxY; y++) {
+                comparisonResult[x][y] = false;
+            }
+        }
+    }
+
+    /**
+     * Build frame for displaying difference using DFS
+     *
+     * @param x          horizontal constraint
+     * @param y          vertical constraint
+     * @param w          width of initial picture
+     * @param h          height of initial picture
+     * @param comparison boolean matrix with difference
+     */
+    private Box buildBox(int x, int y, int w, int h, boolean[][] comparison) {
+        Box box = new Box(x, y);
+        Deque<AbstractMap.SimpleImmutableEntry<Integer, Integer>> stack
+                = new ArrayDeque<>();
+        stack.push(new AbstractMap.SimpleImmutableEntry<>(x, y));
+        while (!stack.isEmpty()) {
+            AbstractMap.SimpleImmutableEntry<Integer, Integer> current = stack.pop();
+            x = current.getKey();
+            y = current.getValue();
+            if (x < 0 || x >= w || y < 0 || y >= h || !comparison[x][y]) {
+                continue;
+            }
+            comparison[x][y] = false;
+            updateBoxSize(box, x, y);
+            stack.push(new AbstractMap.SimpleImmutableEntry<>(x - 1, y));
+            stack.push(new AbstractMap.SimpleImmutableEntry<>(x, y - 1));
+            stack.push(new AbstractMap.SimpleImmutableEntry<>(x + 1, y));
+            stack.push(new AbstractMap.SimpleImmutableEntry<>(x, y + 1));
+        }
+        return box;
+    }
+
+    //Left recursive function for history
+    //(In real project should remove old code)
+/*    private void buildBox(int x, int y, int w, int h, boolean[][] pix, Box box) {
```

Comment:
remove commented code, if it is not used

Reply:
Yes, all comments should be deleted in projects. I wrote this to you on Skype.  But this project is for education and I left it only for my own purposes and for future investigation. I changed the reason in comment.